cmd/gcs-sidecar/main.go

```diff
@@ -214,13 +215,25 @@ func main() {
 		return
 	}
 
+	var snpMode = false
+	if err := pspdriver.StartPSPDriver(ctx); err != nil {
+		// When error happens, pspdriver.GetPspDriverError() returns true.
+		// When it happens, gcs-sidecar should keep the initial "deny" policy
+		// and reject all requests from the host.
+		logrus.WithError(err).Errorf("failed to start PSP driver")
+	} else {
+		snpMode, err = pspdriver.IsSNPMode(ctx)
+		if err != nil {
+			logrus.WithError(err).Errorf("failed to check SNP mode: %v", err)
+		}
+		logrus.Tracef("SNP mode: %v", snpMode)
+	}
+
 	// gcs-sidecar can be used for non-confidentail hyperv wcow
-	// as well. So we do not always want to check for initialPolicyStance
+	// as well. So we do not always want to check for initialPolicyStance.
+	// While it's used only for confidential cwow, always use "deny" as initial policy.
 	var initialEnforcer securitypolicy.SecurityPolicyEnforcer
-	// TODO (kiashok/Mahati): The initialPolicyStance is set to allow
-	// only for dev. This will eventually be set to allow/deny depending on
-	// on whether SNP is supported or not.
-	initialPolicyStance := "allow"
+	initialPolicyStance := "deny"
```

Comment:
The original plan was to do the following (https://github.com/microsoft/hcsshim/pull/2518/commits/0afe43ac9f65b9c0d3c5781c92a579b44308200c). Then Ken decided to always use "deny". I didn't have enough time to ask why. If anyone has major objections, I think we can stick to the original way for now, then come back to it later.

```go
initialPolicyStance := "allow"
	if pspdriver.GetPspDriverError() != nil || snpMode {
		// If the driver failed to start, policy should keep returning "deny" for anything.
		// For SNP environment, the initial policy is "deny" but it will be updated
		// per the user's security policy.
		initialPolicyStance = "deny"
	}
```